app/validators/different_to_from_location_validator.rb

```diff
@@ -0,0 +1,10 @@
+class DifferentToFromLocationValidator < ActiveModel::Validator
+  def validate(record)
+    return if record.to_location_id.blank?
+    return if record.from_location_id.blank?
+
+    if record.to_location_id == record.from_location_id
```

Comment:
Minor, but to keep with the guard theme I'd `return if record.to_location_id != record.from_location_id`

Reply:
Good point, I'll amend the commit.